### PR TITLE
Native Tool Calling and Model Registry Cleanup and Better Runtime Info

### PR DESCRIPTION
**Pull request by Ninot1Quyi**
## Verified functionality

- Verified single-agent survival and construction flows.
- Verified voice broadcast / narration behavior.
- Verified multiple agents can run concurrently while cooperating with each other.
- Tested and passed mainstream provider protocols: OpenAI Chat, OpenAI Responses, Gemini, and Anthropic-compatible messages.
- Verified both user-sent messages and agent-to-agent messages still hit the conversation cache after branching/interruption.

## Update
- Adds native function/tool calling for agent actions, including structured tool calls/results in conversation history.
- Adds native ChatGPT Codex login support, enabling the Codex/ChatGPT Responses transport through a ChatGPT account login flow.
- Adds provider-registry based LLM configuration via `settings_llm_providers.json`, including Kimi, MiniMax, OpenRouter, Gemini, Codex, Replicate, and other presets.
- Splits editable default prompts into markdown files under `profiles/defaults/prompts/` for easier review and prompt maintenance.
- Improves message priority handling so user/admin messages interrupt active actions before the next model turn.
- Makes `newAction`/coding requests independent from the main conversation request path so coding does not pollute conversation cache/state.
- Adds bot-to-bot response branching (`botResponder`) that forks the current conversation context instead of rebuilding or replacing the system prompt.
- Expands Runtime UI trace rendering for native tool calls, tool results, branch decisions, token/cache metadata, and active tool status.

This branch has merged the following 4 PRs:
https://github.com/mindcraft-bots/mindcraft/pull/680
https://github.com/mindcraft-bots/mindcraft/pull/756
https://github.com/mindcraft-bots/mindcraft/pull/752
https://github.com/mindcraft-bots/mindcraft/pull/744
Merging this branch will resolve a total of 5 PRs, including the current one.

**Assisted by Codex GPT-5.5 xhigh & Oh-My-Codex, and has already gone through code quality checks and optimization.** 

<p align="center">
  8160f62b-63f4-4892-835b-fbf1e06ca38f
  dd4fadef-e054-45de-ba59-3e24abac128e
</p>

More stronger compact and ui

2ee032d8-dd8c-4ca9-8de0-208f646dde7c

### CODE REVIEW

File: src/process/agent_process.js
Comment (Ninot1Quyi) [inline src/process/agent_process.js:1]: When restarting, do not inject the init message; only inject it on the first start‑up.

File: profiles/defaults/prompts/conversing.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+You are an AI Minecraft bot named $NAME that can converse with players, see, move, mine, build, and interact with the world by using native tools/function calls.
```
Comment (uukelele) [inline profiles/defaults/prompts/conversing.md]:
```suggestion
You are an AI Minecraft bot named $NAME that can converse with players, see, move, mine, build, and interact with the world by using tools/function calls.
```

File: profiles/tasks/cooking_profile.json [hunk 2]
```diff
@@ -1,11 +1,12 @@
 {
-    "name": "andy", 
-    "model": "claude-3-5-sonnet-latest", 
+    "name": "andy",
+    "model": "claude-3-5-sonnet-latest",
```
Comment (uukelele) [inline profiles/tasks/cooking_profile.json]:
```suggestion
    "model": "claude-opus-4-7",
```

Comment (uukelele) [inline profiles/tasks/cooking_profile.json]:
```suggestion
    "model": "claude-opus-4-7",
```

File: profiles/tasks/crafting_profile.json [hunk 3]
```diff
@@ -1,71 +1,11 @@
 {
-    "name": "andy", 
-    "model": "claude-3-5-sonnet-latest", 
+    "name": "andy",
+    "model": "claude-3-5-sonnet-latest",
```
Comment (uukelele) [inline profiles/tasks/crafting_profile.json]:
```suggestion
    "model": "claude-opus-4-7",
```

File: profiles/andy-4-reasoning.json [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
-    
-    "bot_responder": "You are a minecraft bot named $NAME that is currently in conversation with another AI bot. Both of you can take actions with the !command syntax, and actions take time to complete. You are currently busy with the following action: '$ACTION' but have received a new message. Decide whether to 'respond' immediately or 'ignore' it and wait for your current action to finish. Be conservative and only respond when necessary, like when you need to change/stop your action, or convey necessary information. Example 1: You:Building a house! !newAction('Build a house.').\nOther Bot: 'Come here!'\nYour decision: ignore\nExample 2: You:Collecting dirt !collectBlocks('dirt',10).\nOther Bot: 'No, collect some wood instead.'\nYour decision: respond\nExample 3: You:Coming to you now. !goToPlayer('billy',3).\nOther Bot: 'What biome are you in?'\nYour decision: respond\nActual Conversation: $TO_SUMMARIZE\nDecide by outputting ONLY 'respond' or 'ignore', nothing else. Your decision:"
-
+    "model": {
+        "provider": "ollama_local",
```
Comment (uukelele) [inline profiles/andy-4-reasoning.json]: @Sweaterdog recommends using LMStudio instead of Ollama for local Andy models.

File: profiles/andy-4-reasoning.json [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
+        "model": "sweaterdog/andy-4:micro-q8_0"
```
Comment (uukelele) [inline profiles/andy-4-reasoning.json]: Maybe switch to [andy 4.2](https://huggingface.co/Mindcraft-CE/Andy-4.2)?

File: profiles/andy-4.json [hunk 6]
```diff
@@ -1,7 +1,19 @@
 {
     "name": "andy_4",
-
-    "model": "ollama/sweaterdog/andy-4:micro-q8_0",
-    
-    "embedding": "ollama"
+    "model": {
+        "provider": "ollama_local",
```
Comment (uukelele) [inline profiles/andy-4.json]: Same notes as [above](https://github.com/mindcraft-bots/mindcraft/pull/769/changes#r3178466988)

File: profiles/azure.json [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
+        "model": "gpt-5-nano"
     },
     "embedding": {
-        "api": "azure",
-        "url": "https://<your-resource>.openai.azure.com",
-        "model": "<embedding-deployment-name>",
-        "params": { 
-            "apiVersion": "2024-08-01-preview" 
-        }
+        "provider": "",
```
Comment (uukelele) [inline profiles/azure.json]: Should these not be filled in?

File: profiles/cerebras.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "cerebras",
+    "model": {
+        "provider": "cerebras",
+        "model": "gpt-oss-120b"
+    },
+    "embedding": {
+        "provider": "",
```
Comment (uukelele) [inline profiles/cerebras.json:8]: Same as above, should these not be filled in?

File: profiles/replicate.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "replicate",
+    "model": {
+        "provider": "replicate",
+        "model": "google/gemini-2.5-flash"
```
Comment (uukelele) [inline profiles/replicate.json]:
```suggestion
        "model": "google/gemini-flash-latest"
```

File: scripts/smoke/dev_native_tool_loop.js
Comment (uukelele) [inline scripts/smoke/dev_native_tool_loop.js:1]: Is this left over from your local testing setup? I noticed you [tried to remove it](https://github.com/mindcraft-bots/mindcraft/pull/769/commits/e129513812feeab93825f9b87a9c2c59ef7f9c04)

**Review by uukelele — COMMENTED** [review]
Only managed to view 50/118 files. But overall, I think it is good but I have left some comments for improvement. I will also approve / request more changes when I am able to test this locally.

**Comment by Ninot1Quyi** [comment]
> Only managed to view 50/118 files. But overall, I think it is good but I have left some comments for improvement. I will also approve / request more changes when I am able to test this locally.

Thanks for your review！！！

File: profiles/andy-4.json (hunk 6 above)
Comment (Ninot1Quyi) [reply profiles/andy-4.json]: Merged https://github.com/mindcraft-bots/mindcraft/pull/744 into this PR.

File: profiles/andy-4-reasoning.json (hunk 4 above)
Comment (Ninot1Quyi) [reply profiles/andy-4-reasoning.json]: Merged https://github.com/mindcraft-bots/mindcraft/pull/744 into this PR.
Add a new file and remove 2 file:
git diff: https://github.com/mindcraft-bots/mindcraft/pull/769/changes#diff-ec3ea5289a7dac7b186836f2c147db7575e0a214ed1c77f880d0e42236b77e27

File: profiles/cerebras.json (hunk 8 above)
Comment (Ninot1Quyi) [reply profiles/cerebras.json:8]: These should not be filled in~

Preset templates should not depend on any vendor other than this one, unless the user has filled it in themselves.

Here, the Cerebras vendor does not provide an embedding service. Users should configure the embedding in settings_llm_providers.example.json and then select it here. If this field is left empty, text similarity will be used by default. 

**If this have some free embedding model service, can filled it in.But i have not found**

File: scripts/smoke/dev_native_tool_loop.js
Comment (Ninot1Quyi) [reply scripts/smoke/dev_native_tool_loop.js:1]: yeah, it should be removed

File: profiles/azure.json (hunk 7 above)
Comment (Ninot1Quyi) [reply profiles/azure.json]: yeah, I fixed it

File: profiles/cerebras.json (hunk 8 above)
Comment (uukelele) [reply profiles/cerebras.json:8]: I feel instead of using empty strings you might as well just not have the `embedding` field present in the JSON.

**Comment by masterpatrickpl-coder** [comment]
Happy to test this if needed.

**Comment by uukelele** [comment]
> Happy to test this if needed.

Yeah you can try to test this and feedback the results. See if there are any bugs or edge cases that worked before.

**Comment by Ninot1Quyi** [comment]
> Happy to test this if needed.

Thanks so much! If you find any issues, please feel free to pull this branch and make fixes on top of it. If you need test keys, you can reach me on Discord or by email, and I’d be happy to provide access to some model subscriptions for testing.

**Review by Sweaterdog — APPROVED** [review]
Finally got around to this. It all looks good to me. Nicely done!
